google/cloud/spanner_dbapi/client_side_statement_executor.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 20203 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from google.cloud.spanner_dbapi.parsed_statement import (
+    ParsedStatement,
+    ClientSideStatementType,
+)
+
+
+class StatementExecutor(object):
```

Comment:
Does this benefit from being a class?  (Is `execute()` going to be called more than once?  Will it be called in a context where the connection is not plumbed through but the StatementExecutor object is available?)

There's nothing necessarily wrong with its being a class.  But in Python, if a collection of functions don't have shared state, it's also totally fine to leave them as bare functions, and use the file/module itself as a way of grouping them.  At a quick glance, this class as currently implemented looks to me like it could just be an `execute()` function that takes a `connection` object as an additional argument.  If you have bigger future plans for this class, though, then keeping it as a class makes total sense.

Reply:
Yes I was also confused about making it as a class as I dont see any state that it needs to maintain. My thought was that as I will implement different statements I might need attributes other than connection but I am not sure if they would be needed.
I can remove the class and later add if needed

Reply:
Created b/312624765 for tracking

Reply:
Done